api/src/test/java/org/openmrs/module/ModuleUtilTest.java

```diff
@@ -705,7 +708,52 @@ public void expandJar_shouldExpandFileWithParentTreeIfNameIsFileAndKeepFullPathI
 
 		FileUtils.deleteDirectory(destinationFolder);
 	}
+	
+	/**
+	* @see ModuleUtil#file2url(File)
+	*/
+	@Test
+	public void file2url_shouldReturnNullIfFileIsNull() throws MalformedURLException {
+		URL nullURL = ModuleUtil.file2url(null);
+		Assert.assertNull(nullURL);
+	}
 
+	/**
+	* @see ModuleUtil#file2url(File)
+	*/
+	@Test(expected = MalformedURLException.class)
+	public void file2url_shouldThrowMalformedURLExceptionIfMalformedFilePath() throws MalformedURLException {
+		ModuleUtil.file2url(new File("org/openmrs/" + "\0" + "/include/test1-1.0-SNAPSHOT.omod"));	
+	}
+	
+	/**
+	* @see ModuleUtl#getPackagesFromFile(File)
+	*/
+	@Test
+	public void getPackagesFromFile_shouldReturnEmptyStringSetIfNonJarFile() {
+		File f = new File(this.getClass().getResource("/org/openmrs/module/include/test1-1.0-SNAPSHOT.omod").getFile());
+		Collection<String> packageCollection = ModuleUtil.getPackagesFromFile(f);
+		Assert.assertTrue(packageCollection.isEmpty());
+	}
+	
+	/**
+	* @see ModuleUtl#getPackagesFromFile(File)
+	*/
+	@Test
+	public void getPackagesFromFile_shouldSkipOptionalFoldersIfJarFile() throws IOException{
+		File f = new File(this.getClass().getResource("/org/openmrs/module/include/test1-1.0-SNAPSHOT.omod").getFile());
+		File d = new File("/tmp/test1-1.0-SNAPSHOT.jar");
+		FileUtils.copyFile(f, d);
+		Collection<String> packageCollection = ModuleUtil.getPackagesFromFile(d);
+		
+		Assert.assertFalse(packageCollection.isEmpty());
+		for (String string : packageCollection) {
+			Assert.assertFalse(string.contains("lib"));
+			Assert.assertFalse(string.contains("META-INF"));
+			Assert.assertFalse(string.contains("web.module"));
```

Comment:
Can you point me to the line where ModuleUtil deals with a folder named "web.module"?

Reply:
Thank you @dkayiwa for this catch, it was a typo issue. But here is [the line](https://github.com/openmrs/openmrs-core/blob/4a0feb8da351088f25fdc4e6d324a1f277aa3410/api/src/main/java/org/openmrs/module/ModuleUtil.java#L1141) nearly related to what you asked. I am refactoring this to correct my mistake.